data-fetching-astro-glob (#1255)
withastro/docs: src/pages/en/guides/data-fetching.md

i18nReady: true
---

> `.astro` files can fetch data from your local files, or remote data at build time to help generate your pages.

Reviewer: This sentence feels a bit wordy to me, what do you think of rephrasing it this way? I omitted local *files* but since the example mentions that we are fetching from a local file I think readers aren't going to be confused.
```suggestion
`.astro` files can fetch local and remote data at build time to help generate your pages.
```

## Fetch data from your local files

Get data from one or many local files in your project using Astro's top-level await and [`Astro.glob()`](/en/reference/api-reference/#astroglob). This function returns an array (even if you only query one file!) and gives your `.astro` file access to exported properties like Markdown/MDX frontmatter and more!

```astro "Astro.glob('../pages/post/*.md')"
---
// src/components/blog-archive.astro
const posts = await Astro.glob('../pages/posts/*.md');
---
  <ul>
    {posts.map((post) => (

>       <li><a href={post.frontmatter.url}>{post.frontmatter.title}</li></p>

Author: ```suggestion
      <li><a href={post.frontmatter.url}>{post.frontmatter.title}</a></li>
```
